src/ResourceManager/Compute/Commands.Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs

```diff
@@ -116,23 +125,138 @@ protected PSArgument[] CreateVirtualMachineScaleSetCreateOrUpdateParameters()
     [OutputType(typeof(PSVirtualMachineScaleSet))]
     public partial class NewAzureRmVmss : ComputeAutomationBaseCmdlet
     {
+        public const string SimpleParameterSet = "SimpleParameterSet";
+
         public override void ExecuteCmdlet()
         {
-            ExecuteClientAction(() =>
+            switch (ParameterSetName)
+            {
+                case SimpleParameterSet:
+                    this.StartAndWait(SimpleParameterSetExecuteCmdlet);
+                    break;
+                default:
+                    ExecuteClientAction(() =>
+                    {
+                        if (ShouldProcess(this.VMScaleSetName, VerbsCommon.New))
+                        {
+                            string resourceGroupName = this.ResourceGroupName;
+                            string vmScaleSetName = this.VMScaleSetName;
+                            VirtualMachineScaleSet parameters = new VirtualMachineScaleSet();
+                            ComputeAutomationAutoMapperProfile.Mapper.Map<PSVirtualMachineScaleSet, VirtualMachineScaleSet>(this.VirtualMachineScaleSet, parameters);
+
+                            var result = VirtualMachineScaleSetsClient.CreateOrUpdate(resourceGroupName, vmScaleSetName, parameters);
+                            var psObject = new PSVirtualMachineScaleSet();
+                            ComputeAutomationAutoMapperProfile.Mapper.Map<VirtualMachineScaleSet, PSVirtualMachineScaleSet>(result, psObject);
+                            WriteObject(psObject);
+                        }
+                    });
+                    break;
+            }
+        }
+
+        async Task SimpleParameterSetExecuteCmdlet(IAsyncCmdlet asyncCmdlet)
+        {
+            ResourceGroupName = ResourceGroupName ?? VMScaleSetName;
+            VirtualNetworkName = VirtualNetworkName ?? VMScaleSetName;
+            SubnetName = SubnetName ?? VMScaleSetName;
+            PublicIpAddressName = PublicIpAddressName ?? VMScaleSetName;
+            DomainNameLabel = DomainNameLabel ?? (VMScaleSetName + ResourceGroupName).ToLower();
+            SecurityGroupName = SecurityGroupName ?? VMScaleSetName;
+            LoadBalancerName = LoadBalancerName ?? VMScaleSetName;
+            FrontendPoolName = FrontendPoolName ?? VMScaleSetName;
+            BackendPoolName = BackendPoolName ?? VMScaleSetName;
+
+            // get image
+            var image = Images
+                .Instance
+                .Select(osAndMap =>
+                    new { OsType = osAndMap.Key, Image = osAndMap.Value.GetOrNull(ImageName) })
+                .First(osAndImage => osAndImage.Image != null);
```

Comment:
Is this guaranteed not to throw (that is, there is always an Image that satisfies these criteria)?  If it can throw, we should use FirstOrDefault and throw a meaningful exception is the value is null.

Reply:
fixed.